RequestManagementSystem/Client/ReqClient.py

```diff
@@ -304,17 +304,27 @@ def finalizeRequest(self, requestID, jobID, useCertificates=True):
 
     # The request is 'Done', let's update the job status. If we fail, we should re-try later
     monitorServer = RPCClient("WorkloadManagement/JobMonitoring", useCertificates=useCertificates)
-    res = monitorServer.getJobPrimarySummary(int(jobID))
+    res = monitorServer.getJobSummary(int(jobID))
     if not res["OK"]:
       self.log.error("finalizeRequest: Failed to get job status", "JobID: %d" % jobID)
       return S_ERROR("finalizeRequest: Failed to get job %d status" % jobID)
     elif not res['Value']:
       self.log.info("finalizeRequest: job %d does not exist (anymore): finalizing" % jobID)
       return S_OK()
     else:
-      jobStatus = res["Value"]["Status"]
-      newJobStatus = jobStatus
+      jobStatus = res["Value"]['Status']
       jobMinorStatus = res["Value"]["MinorStatus"]
+      jobAppStatus = ''
+      newJobStatus = ''
+      if jobStatus == 'Stalled':
+        # If job is stalled, find the previous status from the logging info
+        res = monitorServer.getJobLoggingInfo(int(jobID))
+        if not res['OK']:
+          self.log.error("finalizeRequest: Failed to get job logging info", "JobID: %d" % jobID)
+          return S_ERROR("finalizeRequest: Failed to get job %d logging info" % jobID)
+        if len(res['Value']) >= 2:
+          jobStatus, jobMinorStatus, jobAppStatus = res['Value'][-2][:3]
```

Comment:
This basically assumes that what is in `[-1]` is `Stalled`. This is of course completely logic but... can the code check it? Given our record, I don't trust ourselves on this...

Reply:
check added